src/exporter.js

```diff
@@ -64,101 +45,144 @@ export function handler (input, context, callback) {
+      })
+      .on('data', function (data) {
+        let subscriptionName = data[SUBSCRIPTION_NAME]
+        if (!holidaySuspensions.has(subscriptionName)) {
+          let outputCsvRow = {}
+          outputCsvRow[CUSTOMER_REFERENCE] = subscriptionName
+          outputCsvRow[CUSTOMER_TOWN] = data[CITY]
+          outputCsvRow[CUSTOMER_POSTCODE] = formatPostCode(data[POSTAL_CODE])
+          outputCsvRow[CUSTOMER_ADDRESS_LINE_1] = data[ADDRESS_1]
+          outputCsvRow[CUSTOMER_ADDRESS_LINE_2] = data[ADDRESS_2]
+          outputCsvRow[CUSTOMER_FULL_NAME] = data[FIRST_NAME] + ' ' + data[LAST_NAME]
+          outputCsvRow[DELIVERY_QUANTITY] = data[QUANTITY]
+          outputCsvRow[SENT_DATE] = sentDate
+          outputCsvRow[DELIVERY_DATE] = formattedDeliveryDate
+          outputCsvRow[CHARGE_DAY] = chargeDay
+          outputCsvRow[CUSTOMER_PHONE] = data[WORK_PHONE]
+          outputCsvRow[ADDITIONAL_INFORMATION] = data[DELIVERY_INSTRUCTIONS]
+          writeCSVStream.write(outputCsvRow)
+        }
+      })
+      .on('end', function () {
+        writeCSVStream.end()
+      })
+
+    downloadStream.on('error', function (err) {
+      reject(new Error(`error reading holidaySuspensions: ${err}`))
+    })
+      .pipe(csvStream)
+
+    let dateSuffix = deliveryDate.format('DD_MM_YYYY')
     let outputFileName = `HOME_DELIVERY_${chargeDay}_${dateSuffix}.csv`
     let outputLocation = `${stage}/fulfilment_output/${outputFileName}`
-    console.log(`saving fulfilment file to ${BUCKET}/${outputLocation}`)
-    let params = {
-      Bucket: BUCKET,
-      Key: outputLocation,
-      Body: writeCSVStream,
-      ServerSideEncryption: 'aws:kms'
-    }
-    s3.upload(params).send(function (err, data) {
+
+    upload(writeCSVStream, outputLocation, function (err, data) {
       if (err) {
         console.log('ERROR ' + err)
-        callback(err)
+        reject(err)
       } else {
-        callback(null, {
-          fulfilmentFile: outputFileName
-        })
+        resolve(outputFileName)
       }
     })
-  }
+  })
+}
+
+function getStage () {
+  return new Promise((resolve, reject) => {
+    let stage = process.env.Stage
+    if (!stage) {
+      reject(new Error('missing stage env variable'))
+      return
+    }
 
-  console.log(`stage is ${stage}`)
+    if (stage !== 'CODE' && stage !== 'PROD') {
+      reject(new Error(`invalid stage: ${stage}, please fix Stage env variable`))
+      return
+    }
+    console.log(`stage is ${stage}`)
+    resolve(stage)
+  })
+}
+
+function getDeliveryDate (input) {
+  return new Promise((resolve, reject) => {
+    let deliveryDate = moment(input.deliveryDate, 'YYYY-MM-DD')
+    console.log('delivery date was ' + deliveryDate)
+    if (deliveryDate.isValid()) {
+      resolve(deliveryDate)
+    } else {
+      reject(new Error('invalid deliverydate expected format YYYY-MM-DD'))
+    }
+  })
+}
+
+async function asyncHandler (input) {
+  let stage = await getStage()
```

Comment:
I don't really need to download the config from s3 for this lambda but It might make sense to do some useless processing and downloading just to keep it consistent with the other lambdas